source/egg/math/Matrix.cc

```diff
@@ -292,6 +292,34 @@ void Matrix34f::inverseTo33(Matrix34f &out) const {
     out[1, 1] = (mtx[0][0] * mtx[2][2] - mtx[2][0] * mtx[0][2]) * invDet;
 }
 
+/// @addr{0x80199FC8}
+void Matrix34f::ps_inverse(Matrix34f &out) const {
+    f32 fVar14 = fms(mtx[0][1], mtx[1][2], mtx[1][1] * mtx[0][2]);
+    f32 fVar15 = fms(mtx[1][1], mtx[2][2], mtx[2][1] * mtx[1][2]);
+    f32 fVar13 = fms(mtx[2][1], mtx[0][2], mtx[0][1] * mtx[2][2]);
+    f32 determinant = fma(mtx[2][0], fVar14, fma(mtx[1][0], fVar13, mtx[0][0] * fVar15));
+
+    if (determinant == 0.0f) {
+        return;
+    }
+
+    f32 invDet = 1.0f / determinant;
+    f32 scalar = -fms(determinant, invDet * invDet, invDet + invDet);
```

Comment:
```suggestion
    invDet = -fms(determinant, invDet * invDet, invDet + invDet);
```

Given by the refinement (suppose n is the determinant, x is the inverse determinant):
```math
-(n \cdot x \cdot x - (x + x))
= -(nx^2 - 2x)
= -nx^2 + 2x
= x(2 - nx)
```
Matching:
```math
x_{i+1} = x_i(2-nx_i)
```
as a known refinement for 1/n.